src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionUpToDateChecker.cs

```diff
@@ -30,22 +35,25 @@ public void ReportStatus(IReadOnlyList<RestoreSummary> restoreSummaries)
                 {
                     var packageSpec = _cachedDependencyGraphSpec.GetProjectSpec(summary.InputPath);
                     GetOutputFilePaths(packageSpec, out string assetsFilePath, out string cacheFilePath, out string targetsFilePath, out string propsFilePath, out string lockFilePath);
+                    var messages = !packageSpec.RestoreSettings.HideWarningsAndErrors && summary.Errors.Count > 0 ?
+                            summary.Errors :
+                            null;
 
-                    _outputWriteTimes[summary.InputPath] = new RestoreOutputData()
+                    _restoreData[summary.InputPath] = new RestoreData()
                     {
                         _lastAssetsFileWriteTime = GetLastWriteTime(assetsFilePath),
                         _lastCacheFileWriteTime = GetLastWriteTime(cacheFilePath),
                         _lastTargetsFileWriteTime = GetLastWriteTime(targetsFilePath),
                         _lastPropsFileWriteTime = GetLastWriteTime(propsFilePath),
                         _lastLockFileWriteTime = GetLastWriteTime(lockFilePath),
-                        _globalPackagesFolderCreationTime = GetCreationTime(packageSpec.RestoreMetadata.PackagesPath)
+                        _globalPackagesFolderCreationTime = GetCreationTime(packageSpec.RestoreMetadata.PackagesPath),
```

Comment:
Here assigning value to `_globalPackagesFolderCreationTime `, but where are we using it?

Reply:
In AreOutputsUpToDate, https://github.com/NuGet/NuGet.Client/blob/888a672dfa45dcb26dfb348b1247b73117731c7e/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionUpToDateChecker.cs#L231-L246